src/test/app/MultiSign_test.cpp

```diff
@@ -1068,6 +1068,60 @@ class MultiSign_test : public beast::unit_test::suite
         env(noop(alice), msig(becky, demon), fee(3 * baseFee), ter(tefNOT_MULTI_SIGNING));
     }
 
+    void test_multisigningMultisigner()
+    {
+        // Set up a signer list where one of the signers has both the
+        // master disabled and no regular key (because that signer is
+        // exclusively multisigning).  That signer should no longer be
+        // able to successfully sign the signer list.
+
+        using namespace jtx;
+        Env env (*this);
+        Account const alice {"alice", KeyType::ed25519};
+        Account const becky {"becky", KeyType::secp256k1};
+        env.fund (XRP(1000), alice, becky);
+        env.close();
+
+        // alice sets up a signer list with becky as a signer.
+        env (signers (alice, 1, {{becky, 1}}));
+        env.close();
+
+        // becky sets up her signer list.
+        env (signers (becky, 2, {{bogie, 1}, {demon, 1}}));
+        env.close();
+
+        // Because becky has not (yet) disabled her master key, she can
+        // multisign a transaction for alice.
+        auto const baseFee = env.current()->fees().base;
+        env (noop (alice), msig (becky), fee (2 * baseFee));
+        env.close();
+
+        // Now becky disables her master key.
+        env (fset (becky, asfDisableMaster));
+        env.close();
+
+        // Since becky's master key is disabled she can no longer
+        // multisign for alice.  There is no interface that would allow
+        // becky to multisign for alice.
+        env (noop (alice), msig (becky), fee (2 * baseFee),
+            ter (tefMASTER_DISABLED));
+        env.close();
+
+        // Verify that becky cannot sign with a regular key that she has
+        // not yet enabled.
+        Account const beck {"beck", KeyType::ed25519};
+        env (noop (alice), msig (msig::Reg {becky, beck}), fee (2 * baseFee),
+            ter (tefBAD_SIGNATURE));
+        env.close();
+
+        // Once becky gives herself the regular key, she can sign for alice
+        // using that regular key.
+        env (regkey (becky, beck), msig (bogie, demon), fee (3 * baseFee));
+        env.close();
+
+        env (noop (alice), msig (msig::Reg {becky, beck}), fee (2 * baseFee));
+        env.close();
```

Comment:
What do you think about an additional test showing that becky's signers cannot multisign for alice (I guess using `msig (msig::Reg {becky, bogie})`)?
It might be worth giving becky a single signer, because `msig (msig::Reg {becky, bogie}, msig::Reg {becky, demon})` results in `temINVALID` due to both `Signers` accounts' having the same address.
Instead, we could get the desired `tefBAD_SIGNATURE (A signature is provided for a non-signer.)`